components/validation/validation-custom-mixin.js

```diff
@@ -11,60 +11,49 @@ export const ValidationCustomMixin = superclass => class extends superclass {
 
 	constructor() {
 		super();
-		this._forElement = null;
+
+		this._eventSubscriberController = new EventSubscriberController(this, {},
+			{ eventName: 'd2l-validation-custom-connected' }
+		);
+
+		this._forPropertySubscriberController = new ForPropertySubscriberController(this,
+			{ onUnsubscribe: this._onUnsubscribe.bind(this) },
+			{ forProperty: 'for' }
+		);
 	}
 
 	get forElement() {
-		return this._forElement;
+		if (this.for) {
+			// Validation custom components only support one for element
+			return this._forPropertySubscriberController.providers[0];
```

Comment:
I'd suggest doing an array length check first and return `null` otherwise. Consumers could have been relying on this being `null` if no element was connected and now accessing this property will throw an array index out of bounds exception.